dockerbuild: stop container before committing it

csrwng opened the pull request:
The problem:
Images built with the new dockerbuild command get the wrong permisisons (0700 instead of 0755) in the /run/secrets (or /var/run/secrets) directory when running on non-RedHat docker hosts. This breaks loading secrets in containers created from those images. 

The fix:
I found that this happens only when the container is committed while still running. If I the container is stopped before committing, the permissions for that directory come out right.

csrwng commented:
@smarterclayton ptal
cc @bparees

smarterclayton commented on pkg/util/docker/dockerfile/builder/client.go:
```diff
@@ -19,6 +19,9 @@ import (
 )
 
+// stopTimeoutSeconds is the timeout for stopping the running container
+const stopTimeoutSeconds = 2
```
Should have zero timeout - not sure why we would need it to be graceful.

smarterclayton commented on pkg/util/docker/dockerfile/builder/client.go:
```diff
@@ -175,6 +178,16 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
+	if mustStart {
+		glog.V(4).Infof("Stopping container %s ...", e.Container.ID)
+		if e.LogFn != nil {
+			e.LogFn("Stopping container...")
```
Don't print this - not useful to the user.

smarterclayton commented on pkg/util/docker/dockerfile/builder/client.go:
```diff
@@ -175,6 +178,16 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
+		if e.LogFn != nil {
+			e.LogFn("Stopping container...")
+		}
+		if err := e.Client.StopContainer(e.Container.ID, stopTimeoutSeconds); err != nil {
```
Just use 0 here.

smarterclayton commented:
Before this merges, be sure to fix the conformance_test to remove the "Red Hat secrets injection" workaround and verify it still runs.

csrwng commented:
fixed code and updated conformance test. However, the conformance test is failing for me, but I don't think it's due to this change:
```
--- FAIL: TestConformanceInternal (3.66s)
	conformance_test.go:431: a=map[.dockerignore:0xc2080afee0] b=map[] diff=map[]
	conformance_test.go:328: 0: full Docker build was not equivalent to squashed image metadata fixtures/ignore/Dockerfile
```

taking a closer look

csrwng commented:
I reverted to master and I still get the failure above so yes it doesn't look like it's related to this.

smarterclayton commented:
Internal conformance and external conformance have a mismatch right now.  Can you comment out that test and run the other ones?

csrwng commented:
Commented out the test. Other tests pass.

smarterclayton commented:
Lgtm [merge]
